### PR TITLE
Fix msat balance reporting from alby api

### PR DESCRIPTION
**Pull request by bitkarrot**
Ref https://github.com/lnbits/lnbits/issues/2211#issuecomment-1913219499

- fixed balance with 1000 multiplier for msat conversion
- tested send and receive invoices. 

Note: perhaps we should make a unit test for this, in case API changes in the future. Not sure if there any standard other than the existing tests folder. LMK. 

@dni

### CODE REVIEW

File: lnbits/wallets/alby.py [hunk 1]
```diff
@@ -47,7 +47,8 @@ async def status(self) -> StatusResponse:
         data = r.json()["balance"]
         if r.is_error:
             return StatusResponse(data["error"], 0)
```
Comment (prusnak) [inline lnbits/wallets/alby.py:49]: Unrelated to the PR, but this code looks really weird.

1. First, we parse the JSON, but we check if there is an error only afterwards.
2. What's even weirder is that we access `data["error"]`, but I don't think `data` is a dict at this point.

Maybe this section should be this instead?

I checked the [documentation here](https://guides.getalby.com/developer-guide/v/alby-wallet-api/reference/api-reference/account) but I haven't tested it. 

```suggestion
        if r.is_error:
            error_message = r.json()["message"]
            return StatusResponse(error_message, 0)
        data = r.json()["balance"]
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: lnbits/wallets/alby.py (hunk 1 above)
Comment (bitkarrot) [reply lnbits/wallets/alby.py:49]: you are correct, it should message, and consistent with other methods in same class. 

Should we have unit tests to simulate this testing? Most projects at this size start to have them.

**Comment by bitkarrot** [comment]
having some difficulty connecting to Alby API at the moment. but feel free to validate :)

File: lnbits/wallets/alby.py (hunk 1 above)
Comment (bitkarrot) [reply lnbits/wallets/alby.py]: Same pattern appears in opennode.py

https://github.com/lnbits/lnbits/blob/e1bb2113ed92556c6227f6de14126570f808007a/lnbits/wallets/opennode.py#L59-L63

Comment (bitkarrot) [reply lnbits/wallets/alby.py]: I reviewed the other wallet files (e.g. lnpay, etc). Every single one has a slightly different style. Not sure what you guys want to do for handling errors and exceptions, so I originally templated the alby wallet integration off of opennode.py

Comment (prusnak) [reply lnbits/wallets/alby.py]: > Same pattern appears in opennode.py

Can you please fix opennode in this PR the same way like you fixed alby?

**Comment by theroyalwhale** [comment]
when merge? :)

**Comment by prusnak** [comment]
I cherry picked your changes into https://github.com/lnbits/lnbits/pull/2221 where I also fixed the same issues in opennode

Thanks!

**Comment by bitkarrot** [comment]
@prusnak ok! You beat me to it while I was sleeping! :D

**Pull request closed without merging**